src/libs/blueprint/conduit_blueprint_mesh_examples.cpp

```diff
@@ -2532,6 +2532,389 @@ void spiral(index_t ndoms,
+
+    float64_array cir_a = res["fields/circle_a/values"].value();
+    float64_array cir_b = res["fields/circle_b/values"].value();
+    float64_array cir_c = res["fields/circle_c/values"].value();
+
+    float64_array mat_bg = res["matsets/matset/volume_fractions/background"].value();
+    float64_array mat_ca = res["matsets/matset/volume_fractions/circle_a"].value();
+    float64_array mat_cb = res["matsets/matset/volume_fractions/circle_b"].value();
+    float64_array mat_cc = res["matsets/matset/volume_fractions/circle_c"].value();
+    
+    for(index_t idx = 0; idx < elements; idx++)
+    {
+        mat_ca[idx] = cir_a[idx];
+        mat_cb[idx] = cir_b[idx];
+        mat_cc[idx] = cir_c[idx];
+        mat_bg[idx] = 1. - (cir_a[idx] + cir_b[idx] + cir_c[idx]);
+    }
+}
+
+void make_sparse(Node & src, index_t len, Node & n)
+{
+    float64_array src_val = src.value();
+
+    index_t nz = 0;
+    for (index_t idx = 0; idx < len; ++idx)
+    {
+        if (src_val[idx] > 0)
+        {
+            nz += 1;
+        }
+    }
+
+    n["values"].set(DataType::float64(nz));
+    n["element_ids"].set(DataType::int32(nz));
+    float64_array n_val = n["values"].value();
+    int32_array n_element_ids = n["element_ids"].value();
+
+    index_t nidx = 0;
+    for (index_t idx = 0; idx < len; ++idx)
+    {
+        if (src_val[idx] > 0)
+        {
+            n_val[nidx] = src_val[idx];
+            n_element_ids[nidx] = idx;
+            nidx += 1;
+        }
+    }
+}
+
+void venn_sparse_by_material_matset(Node &res,
+                                    index_t nx,
+                                    index_t ny)
+{
+    // create the materials
+
+    float64_array cir_a = res["fields/circle_a/values"].value();
+    float64_array cir_b = res["fields/circle_b/values"].value();
+    float64_array cir_c = res["fields/circle_c/values"].value();
+
+    res["matsets/matset/topology"] = "topo";
+    Node &bg = res["matsets/matset/volume_fractions/background"];
+    Node &ca = res["matsets/matset/volume_fractions/circle_a"];
+    Node &cb = res["matsets/matset/volume_fractions/circle_b"];
+    Node &cc = res["matsets/matset/volume_fractions/circle_c"];
+    
+    index_t elements = nx * ny;
+
+    make_sparse(res["fields/circle_a/values"], elements, ca);
+
+    make_sparse(res["fields/circle_b/values"], elements, cb);
+
+    make_sparse(res["fields/circle_c/values"], elements, cc);
+
+    // The background material volume fraction depends on the other three
+    // materials, so we deal with it in a custom loop.
+    index_t bgcount = 0;
+    for (index_t idx = 0; idx < elements; ++idx)
+    {
+        if (cir_a[idx] + cir_b[idx] + cir_c[idx] < 1.) bgcount += 1;
+    }
+
+    bg["values"].set(DataType::float64(bgcount));
+    bg["element_ids"].set(DataType::int32(bgcount));
+    float64_array bg_val = bg["values"].value();
+    int32_array bg_idx = bg["element_ids"].value();
+
+    index_t nidx = 0;
```

Comment:
Just like `nidx` refers to the index in the `n` node in `make_sparse`, could we name this instead to `bgidx` since it refers to an index in the `bg` node?

Reply:
I changed the `n` and `n_` names and prefixes to `sparse` and `sparse_`.  I think it is much clearer.